### PR TITLE
Rotate Images to Upright Position in preprocess

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

Closes: #6617

Added a unit test. Test visually by running `image_mod_default_image` with the `rotated_image.jpeg` saved in `test/test_files`.

The correct output is a 45 degree rotation from the upright position.

### This branch
image

### Main
image

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: gradio/processing_utils.py [hunk 1]
```diff
@@ -58,21 +58,6 @@ def extract_base64_data(x: str) -> str:
 #########################
 
 
-def decode_base64_to_image(encoding: str) -> Image.Image:
```
Comment (freddyaboulton) [inline gradio/processing_utils.py:61]: not used anywhere

File: gradio/components/image.py [hunk 2]
```diff
@@ -8,7 +8,8 @@
 
 import numpy as np
 from gradio_client.documentation import document, set_documentation_group
-from PIL import Image as _Image  # using _ to minimize namespace pollution
+from PIL import Image as _Image
+from PIL import ImageOps  # using _ to minimize namespace pollution
```
Comment (whitphx) [inline gradio/components/image.py:12]: nit: This comment should be in the previous line.

Comment (freddyaboulton) [reply gradio/components/image.py:12]: Yea I think black actually did this cause I originally placed both imports in the same line and it complained about length. Will fix!

File: gradio/components/image.py [hunk 3]
```diff
@@ -162,6 +163,13 @@ def preprocess(
             return str(file_path)
 
         im = _Image.open(file_path)
+        exif = im.getexif()
+        # 274 is the code for image rotation and 1 means "correct orientation"
+        if exif.get(274, 1) != 1 and hasattr(ImageOps, "exif_transpose"):
+            try:
+                im = ImageOps.exif_transpose(im)
+            except Exception:
```
[the remark this answers is not in the document]
Comment (abidlabs) [reply gradio/components/image.py:171]: Oh sorry I just realized you catch the exception. But I think `warnings.warn()` would be better than `print()`

**Comment by abidlabs** [comment]
LGTM, just a tiny nit! Thanks @freddyaboulton

**Review by abidlabs — APPROVED** [review]

File: gradio/components/image.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (freddyaboulton) [reply gradio/components/image.py:171]: I'm not raising an exception here - I'm catching it and printing a message instage. Should be the same behavior as `decode_base64_to_image` but without `log.warning`

**Review by dawoodkhan82 — APPROVED** [review]
Tested & lgtm

**Comment by freddyaboulton** [comment]
Thanks so much for the reviews!

**Pull request merged**